File: FrontEnd (C#)/SoftProgBusiness/GestPrestamos/BOI/SancionBOImpl.cs
using SoftProgBusiness.GestPrestamos.BO;
using SoftProgModel.GestPrestamos;
using SoftProgPersistance.GestPrestamos.DAO;
using SoftProgPersistance.GestPrestamos.Impl;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftProgBusiness.GestPrestamos.BOI
{
    public class SancionBOImpl : ISancionBO
    {
        private SancionDAO daoSancion;
        public SancionBOImpl()
        {
            daoSancion = new SancionImpl();
        }
        public int eliminar(int idObjeto)
        {
            return daoSancion.eliminar(idObjeto);   
        }

        public int finalizar_sancion(int id_sancion)
        {
            return daoSancion.finalizar_sancion(id_sancion);
        }

        public int insertar(Sancion objeto)
        {
            return daoSancion.insertar(objeto); 
        }

        public BindingList<Sancion> listarTodos()
        {
            return daoSancion.listarTodos();    
        }

        public int modificar(Sancion objeto)
        {
            return daoSancion.modificar(objeto);    
        }

        public Sancion obtenerPorId(int idObjeto)
        {
            return daoSancion.obtenerPorId(idObjeto);   
        }

        public void validar(Sancion objeto)
        {
            throw new NotImplementedException();
        }
    }
}

File: BackEnd (Java)/BibliotecaBusiness/src/main/java/biblioteca/gestionPrestamo/boImpl/SancionBOImpl.java

package biblioteca.gestionPrestamo.boImpl;

import biblioteca.gestionPrestamo.bo.SancionBO;
import biblioteca.gestionPrestamo.dao.SancionDAO;
import biblioteca.gestionPrestamo.mysql.SancionImpl;
import java.util.ArrayList;
import pe.edu.pucp.utilsarmy.gestion_de_prestamos.model.Sancion;

public class SancionBOImpl implements SancionBO{
private final SancionDAO daoSancion;

    public SancionBOImpl() {
        daoSancion = new SancionImpl();
    }

    @Override
    public int insertar(Sancion objeto) throws Exception {
        return daoSancion.insertar(objeto);
    }

    @Override
    public int modificar(Sancion objeto) throws Exception {
        return daoSancion.modificar(objeto);
    }

    @Override
    public int eliminar(int idObjeto) throws Exception {
        return daoSancion.eliminar(idObjeto);
    }

    @Override
    public Sancion obtenerPorId(int idObjeto) throws Exception {
        return daoSancion.obtenerPorId(idObjeto);
    }

    @Override
    public ArrayList<Sancion> listarTodos() throws Exception {
        return daoSancion.listarTodos();
    }
    @Override
    public void validar(Sancion objeto) throws Exception {
        throw new UnsupportedOperationException("Not supported yet."); // Generated from nbfs://nbhost/SystemFileSystem/Templates/Classes/Code/GeneratedMethodBody
    }

    @Override
    public ArrayList<Sancion> listarPorUsuario(int idUsuario) {
        return daoSancion.listarPorUsuario(idUsuario);
    }
    
}
